sharding/sharding.go

```diff
@@ -212,6 +216,59 @@ func (r *ShardingFerry) deltaCopyJoinedTables() error {
 	return r.Ferry.RunStandaloneDataCopy(tables)
 }
 
+func (r *ShardingFerry) copyPrimaryKeyTables() error {
+	pkTables := map[string]struct{}{}
+	for _, name := range r.config.PrimaryKeyTables {
+		pkTables[name] = struct{}{}
+	}
+
+	r.config.TableFilter.(*ShardedTableFilter).PrimaryKeyTables = pkTables
+	r.config.CopyFilter.(*ShardedCopyFilter).PrimaryKeyTables = pkTables
+
+	sourceDbTables, err := ghostferry.LoadTables(r.Ferry.SourceDB, r.config.TableFilter)
+	if err != nil {
+		return err
+	}
+
+	tables := []*schema.Table{}
+	for _, table := range sourceDbTables.AsSlice() {
+		if _, exists := pkTables[table.Name]; exists {
+			if len(table.PKColumns) != 1 {
+				return fmt.Errorf("Multiple PK columns are not supported with the PrimaryKeyTables option")
+			}
+			tables = append(tables, table)
+		}
+	}
+
+	if len(tables) == 0 {
+		r.logger.Warn("found no primary key tables to copy")
```

Comment:
I assume this is for cases where the "shops table isn't sharded yet"?

Reply:
yeah or the PrimaryKeyTables config is not provided. maybe we should return an error here

Reply:
If we return an error then people who's tenant table isn't sharded can't use this, right? I think we should allow them to.

Reply:
Or do you mean return an error if a table name is specified but the table isn't found? If so then I agree.

Reply:
yup latter is what i meant. will push that change